packages/runtime/container-runtime/src/test/gc/gcTelemetryTracker.spec.ts

```diff
@@ -0,0 +1,737 @@
+		// Path starting with "/_blobs" - blob.
+		// Path with one part such as "/id1" - data stores.
+		// Path with two parts such as "/id1/id2" - sub data stores.
+		// Everything else - other.
+		const getNodeType = (nodePath: string) => {
+			if (nodePath.split("/")[1] === BlobManager.basePath) {
+				return GCNodeType.Blob;
+			}
+			if (nodePath.split("/").length === 2) {
+				return GCNodeType.DataStore;
+			}
+			if (nodePath.split("/").length === 3) {
+				return GCNodeType.SubDataStore;
+			}
+			return GCNodeType.Other;
+		};
+		const tracker = new GCTelemetryTracker(
+			mc,
+			{ inactiveTimeoutMs, sweepTimeoutMs: enableSweep ? sweepTimeoutMs : undefined },
+			isSummarizerClient,
+			false /* gcTombstoneEnforcementAllowed */,
+			{ createContainerRuntimeVersion: pkgVersion },
+			getNodeType,
+			(nodeId: string) => unreferencedNodesState.get(nodeId),
+			async (nodeId: string) => testPkgPath,
+		);
+		return tracker;
+	}
+
+	/**
+	 * For each node in nodeIds, add an entry in `unreferencedNodesState` indicating that the node was
+	 * just unreferenced.
+	 */
+	function markNodesUnreferenced(nodeIds: string[]) {
+		nodeIds.forEach((nodeId) => {
+			unreferencedNodesState.set(
+				nodeId,
+				new UnreferencedStateTracker(
+					Date.now(),
+					inactiveTimeoutMs,
+					Date.now(),
+					sweepTimeoutMs,
+				),
+			);
+		});
+	}
+
+	// Mock node loaded and changed activity for the given nodes.
+	function updateNodes(nodeIds: string[]) {
+		nodeIds.forEach((nodeId) => {
+			telemetryTracker.nodeUsed({
+				nodeId,
+				usageType: "Changed",
+				currentReferenceTimestampMs: Date.now(),
+				packagePath: testPkgPath,
+				completedGCRuns: 0,
+				isTombstoned: false,
+			});
+			telemetryTracker.nodeUsed({
+				nodeId,
+				usageType: "Loaded",
+				currentReferenceTimestampMs: Date.now(),
+				packagePath: testPkgPath,
+				completedGCRuns: 0,
+				isTombstoned: false,
+			});
+		});
+	}
+
+	// Mock node revived activity for the given nodes.
+	function reviveNode(fromId: string, toId: string, isTombstoned = false) {
+		telemetryTracker.nodeUsed({
+			nodeId: toId,
+			usageType: "Revived",
+			currentReferenceTimestampMs: Date.now(),
+			packagePath: testPkgPath,
+			completedGCRuns: 0,
+			isTombstoned,
+			fromId,
+		});
+		unreferencedNodesState.delete(toId);
+	}
+
+	/**
+	 * For summarizer clients, inactive / sweep ready events are not logged when on node usage. They are logged when GC
+	 * runs next time. This emulates that by calling the functions in the telemetry tracker that are called when GC runs.
+	 */
+	async function generateAllEvents(isSummarizerClient: boolean) {
+		if (!isSummarizerClient) {
+			return;
+		}
+		telemetryTracker.logSweepEvents(mc.logger, Date.now(), unreferencedNodesState, 0);
+		await telemetryTracker.logUnreferencedEvents(mc.logger);
+	}
+
+	before(() => {
+		clock = useFakeTimers();
+	});
+
+	beforeEach(() => {
+		mockLogger = new MockLogger();
+		mc = mixinMonitoringContext(
```

Comment:
I think you can do this in `before` not `beforeEach`, not that it matters.

Reply:
It doesn't really matter and its okay to reset them after each test.